### ISSUE TITLE
Specify .pre file as requirement if .pre file does not exist to reduce redundant runs

### ISSUE THREAD

**Issue by user202729** [vectorgraphics/asymptote#421]
Currently, with the following `a.tex` (assume folder is empty apart from latexmkrc):

```tex
%! TEX program = pdflatex
\documentclass{article}
\usepackage[nofiles]{attachfile2}
\usepackage[attach]{asymptote}
\begin{document}
123
\begin{asy}
	label("45");
\end{asy}

\end{document}
```

If you type `latexmk` you get:

<details>

```
Rc files read:
  /home/user202729/.latexmkrc
  .latexmkrc
Latexmk: This is Latexmk, John Collins, 7 Jan. 2023. Version 4.79.
No existing .aux file, so I'll make a simple one, and require run of *latex.
Latexmk: applying rule 'lualatex'...
Rule 'lualatex':  Reasons for rerun
Category 'other':
  Rerun of 'lualatex' forced or previously required

------------
Run number 1 of rule 'lualatex'
------------
------------
Running 'lualatex  -recorder  "a.tex"'
------------
This is LuaHBTeX, Version 1.17.0 (TeX Live 2023/Arch Linux) 
 restricted system commands enabled.
(./a.tex
LaTeX2e <2022-11-01> patch level 1
 L3 programming layer <2023-02-22>
(/usr/share/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/usr/share/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texmf-dist/tex/latex/attachfile2/attachfile2.sty
(/usr/share/texmf-dist/tex/generic/iftex/iftex.sty)
(/usr/share/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texmf-dist/tex/latex/graphics/color.sty
(/usr/share/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/share/texmf-dist/tex/latex/graphics-def/luatex.def)
(/usr/share/texmf-dist/tex/latex/graphics/mathcolor.ltx))
(/usr/share/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
(/usr/share/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty)
(/usr/share/texmf-dist/tex/latex/kvoptions/kvoptions.sty
(/usr/share/texmf-dist/tex/latex/kvsetkeys/kvsetkeys.sty))
(/usr/share/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty)
(/usr/share/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/share/texmf-dist/tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/usr/share/texmf-dist/tex/latex/hycolor/hycolor.sty)
(/usr/share/texmf-dist/tex/latex/letltxmacro/letltxmacro.sty)
(/usr/share/texmf-dist/tex/latex/auxhook/auxhook.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texmf-dist/tex/latex/refcount/refcount.sty)
(/usr/share/texmf-dist/tex/generic/gettitlestring/gettitlestring.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/share/texmf-dist/tex/generic/intcalc/intcalc.sty)
(/usr/share/texmf-dist/tex/generic/etexcmds/etexcmds.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/puenc.def)
(/usr/share/texmf-dist/tex/latex/url/url.sty)
(/usr/share/texmf-dist/tex/generic/bitset/bitset.sty
(/usr/share/texmf-dist/tex/generic/bigintcalc/bigintcalc.sty))
(/usr/share/texmf-dist/tex/latex/base/atbegshi-ltx.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/hluatex.def
(/usr/share/texmf-dist/tex/generic/stringenc/stringenc.sty)
(/usr/share/texmf-dist/tex/latex/base/atveryend-ltx.sty)
(/usr/share/texmf-dist/tex/latex/rerunfilecheck/rerunfilecheck.sty
(/usr/share/texmf-dist/tex/generic/uniquecounter/uniquecounter.sty)))
(/usr/share/texmf-dist/tex/latex/attachfile2/atfi-luatex.def))
(/usr/share/texmf/tex/latex/asymptote/asymptote.sty
(/usr/share/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)))
(/usr/share/texmf-dist/tex/generic/iftex/ifpdf.sty)
(/usr/share/texmf-dist/tex/generic/iftex/ifxetex.sty)
(/usr/share/texmf-dist/tex/generic/catchfile/catchfile.sty))
(/usr/share/texmf-dist/tex/latex/l3backend/l3backend-luatex.def) (./a.aux)
(/usr/share/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/share/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty
(/usr/share/texmf-dist/tex/latex/grfext/grfext.sty)
(/usr/share/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))

Package asymptote Warning: file `a-1.pdf' not found on input line 9.

[1{/home/user202729/.texlive/texmf-var/fonts/map/pdftex/updmap/pdftex.map}]
(./a.aux)

Package rerunfilecheck Warning: File `a.out' has changed.
(rerunfilecheck)                Rerun to get outlines right
(rerunfilecheck)                or use package `bookmark'.

)
 411 words of node memory still in use:
   3 hlist, 1 vlist, 1 rule, 2 glue, 3 kern, 1 glyph, 4 attribute, 49 glue_spec
, 4 attribute_list, 1 write nodes
   avail lists: 2:49,3:9,4:12,5:39,6:3,7:67,9:54
</usr/share/texmf-dist/fonts/opentype/public/lm/lmroman10-regular.otf>
Output written on a.pdf (1 page, 2971 bytes).
Transcript written on a.log.
Latexmk: Getting log file 'a.log'
Latexmk: Examining 'a.fls'
Latexmk: Examining 'a.log'
Latexmk: Missing input file 'a-1.pdf' (or dependence on it) from following:
  Package asymptote Warning: file `a-1.pdf' not found on input line 9.
Latexmk: References changed.
Latexmk: Log file says output to 'a.pdf'
Latexmk: applying rule 'cusdep asy pdf a-1'...
Rule 'cusdep asy pdf a-1':  Reasons for rerun
Changed files or newly in use/created:
  a-1.asy
Category 'no_dest':
  cusdep asy pdf a-1

------------
Run number 1 of rule 'cusdep asy pdf a-1'
------------
Latexmk: applying rule 'lualatex'...
Rule 'lualatex':  Reasons for rerun
Changed files or newly in use/created:
  a-1.pdf
  a.aux
  a.out

------------
Run number 2 of rule 'lualatex'
------------
------------
Running 'lualatex  -recorder  "a.tex"'
------------
This is LuaHBTeX, Version 1.17.0 (TeX Live 2023/Arch Linux) 
 restricted system commands enabled.
(./a.tex
LaTeX2e <2022-11-01> patch level 1
 L3 programming layer <2023-02-22>
(/usr/share/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/usr/share/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texmf-dist/tex/latex/attachfile2/attachfile2.sty
(/usr/share/texmf-dist/tex/generic/iftex/iftex.sty)
(/usr/share/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texmf-dist/tex/latex/graphics/color.sty
(/usr/share/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/share/texmf-dist/tex/latex/graphics-def/luatex.def)
(/usr/share/texmf-dist/tex/latex/graphics/mathcolor.ltx))
(/usr/share/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
(/usr/share/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty)
(/usr/share/texmf-dist/tex/latex/kvoptions/kvoptions.sty
(/usr/share/texmf-dist/tex/latex/kvsetkeys/kvsetkeys.sty))
(/usr/share/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty)
(/usr/share/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/share/texmf-dist/tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/usr/share/texmf-dist/tex/latex/hycolor/hycolor.sty)
(/usr/share/texmf-dist/tex/latex/letltxmacro/letltxmacro.sty)
(/usr/share/texmf-dist/tex/latex/auxhook/auxhook.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texmf-dist/tex/latex/refcount/refcount.sty)
(/usr/share/texmf-dist/tex/generic/gettitlestring/gettitlestring.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/share/texmf-dist/tex/generic/intcalc/intcalc.sty)
(/usr/share/texmf-dist/tex/generic/etexcmds/etexcmds.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/puenc.def)
(/usr/share/texmf-dist/tex/latex/url/url.sty)
(/usr/share/texmf-dist/tex/generic/bitset/bitset.sty
(/usr/share/texmf-dist/tex/generic/bigintcalc/bigintcalc.sty))
(/usr/share/texmf-dist/tex/latex/base/atbegshi-ltx.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/hluatex.def
(/usr/share/texmf-dist/tex/generic/stringenc/stringenc.sty)
(/usr/share/texmf-dist/tex/latex/base/atveryend-ltx.sty)
(/usr/share/texmf-dist/tex/latex/rerunfilecheck/rerunfilecheck.sty
(/usr/share/texmf-dist/tex/generic/uniquecounter/uniquecounter.sty)))
(/usr/share/texmf-dist/tex/latex/attachfile2/atfi-luatex.def))
(/usr/share/texmf/tex/latex/asymptote/asymptote.sty (./a.pre)
(/usr/share/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)))
(/usr/share/texmf-dist/tex/generic/iftex/ifpdf.sty)
(/usr/share/texmf-dist/tex/generic/iftex/ifxetex.sty)
(/usr/share/texmf-dist/tex/generic/catchfile/catchfile.sty))
(/usr/share/texmf-dist/tex/latex/l3backend/l3backend-luatex.def) (./a.aux)
(/usr/share/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/share/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty
(/usr/share/texmf-dist/tex/latex/grfext/grfext.sty)
(/usr/share/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg)) (./a.out)
(./a.out)<<a-1.pdf>> [1{/home/user202729/.texlive/texmf-var/fonts/map/pdftex/up
dmap/pdftex.map}<./a-1.pdf>] (./a.aux))
 411 words of node memory still in use:
   3 hlist, 1 vlist, 1 rule, 2 glue, 3 kern, 1 glyph, 4 attribute, 49 glue_spec
, 4 attribute_list, 1 write nodes
   avail lists: 2:71,3:12,4:12,5:42,6:6,7:91,9:111
</usr/share/texmf-dist/fonts/opentype/public/lm/lmroman10-regular.otf>
Output written on a.pdf (1 page, 21301 bytes).
Transcript written on a.log.
Latexmk: Getting log file 'a.log'
Latexmk: Examining 'a.fls'
Latexmk: Examining 'a.log'
Latexmk: Log file says output to 'a.pdf'
Latexmk: applying rule 'lualatex'...
Rule 'lualatex':  Reasons for rerun
Changed files or newly in use/created:
  a.pre

------------
Run number 3 of rule 'lualatex'
------------
------------
Running 'lualatex  -recorder  "a.tex"'
------------
This is LuaHBTeX, Version 1.17.0 (TeX Live 2023/Arch Linux) 
 restricted system commands enabled.
(./a.tex
LaTeX2e <2022-11-01> patch level 1
 L3 programming layer <2023-02-22>
(/usr/share/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/usr/share/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texmf-dist/tex/latex/attachfile2/attachfile2.sty
(/usr/share/texmf-dist/tex/generic/iftex/iftex.sty)
(/usr/share/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texmf-dist/tex/latex/graphics/color.sty
(/usr/share/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/share/texmf-dist/tex/latex/graphics-def/luatex.def)
(/usr/share/texmf-dist/tex/latex/graphics/mathcolor.ltx))
(/usr/share/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
(/usr/share/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty)
(/usr/share/texmf-dist/tex/latex/kvoptions/kvoptions.sty
(/usr/share/texmf-dist/tex/latex/kvsetkeys/kvsetkeys.sty))
(/usr/share/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty)
(/usr/share/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/share/texmf-dist/tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/usr/share/texmf-dist/tex/latex/hycolor/hycolor.sty)
(/usr/share/texmf-dist/tex/latex/letltxmacro/letltxmacro.sty)
(/usr/share/texmf-dist/tex/latex/auxhook/auxhook.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texmf-dist/tex/latex/refcount/refcount.sty)
(/usr/share/texmf-dist/tex/generic/gettitlestring/gettitlestring.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/share/texmf-dist/tex/generic/intcalc/intcalc.sty)
(/usr/share/texmf-dist/tex/generic/etexcmds/etexcmds.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/puenc.def)
(/usr/share/texmf-dist/tex/latex/url/url.sty)
(/usr/share/texmf-dist/tex/generic/bitset/bitset.sty
(/usr/share/texmf-dist/tex/generic/bigintcalc/bigintcalc.sty))
(/usr/share/texmf-dist/tex/latex/base/atbegshi-ltx.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/hluatex.def
(/usr/share/texmf-dist/tex/generic/stringenc/stringenc.sty)
(/usr/share/texmf-dist/tex/latex/base/atveryend-ltx.sty)
(/usr/share/texmf-dist/tex/latex/rerunfilecheck/rerunfilecheck.sty
(/usr/share/texmf-dist/tex/generic/uniquecounter/uniquecounter.sty)))
(/usr/share/texmf-dist/tex/latex/attachfile2/atfi-luatex.def))
(/usr/share/texmf/tex/latex/asymptote/asymptote.sty (./a.pre)
(/usr/share/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)))
(/usr/share/texmf-dist/tex/generic/iftex/ifpdf.sty)
(/usr/share/texmf-dist/tex/generic/iftex/ifxetex.sty)
(/usr/share/texmf-dist/tex/generic/catchfile/catchfile.sty))
(/usr/share/texmf-dist/tex/latex/l3backend/l3backend-luatex.def) (./a.aux)
(/usr/share/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/share/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty
(/usr/share/texmf-dist/tex/latex/grfext/grfext.sty)
(/usr/share/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg)) (./a.out)
(./a.out)<<a-1.pdf>> [1{/home/user202729/.texlive/texmf-var/fonts/map/pdftex/up
dmap/pdftex.map}<./a-1.pdf>] (./a.aux))
 411 words of node memory still in use:
   3 hlist, 1 vlist, 1 rule, 2 glue, 3 kern, 1 glyph, 4 attribute, 49 glue_spec
, 4 attribute_list, 1 write nodes
   avail lists: 2:71,3:12,4:12,5:42,6:6,7:91,9:111
</usr/share/texmf-dist/fonts/opentype/public/lm/lmroman10-regular.otf>
Output written on a.pdf (1 page, 21301 bytes).
Transcript written on a.log.
Latexmk: Getting log file 'a.log'
Latexmk: Examining 'a.fls'
Latexmk: Examining 'a.log'
Latexmk: Log file says output to 'a.pdf'
Latexmk: All targets () are up-to-date
```

</details>

tl;dr: Latexmk needs 3 runs.

I propose implementing the following change (or something equivalent, I don't know if this is a good idea, but it's [recommended by Ulrike](https://tex.stackexchange.com/questions/685455/latexmk-lots-of-runs-caused-by-files-to-be-created-after-first-latex-run/685662#comment1701136_685455)), so it only need 2 runs: In `asymptote.sty`,

```diff
-\InputIfFileExists{\jobname.pre}{}{}
+\InputIfFileExists{\jobname.pre}{}{\typeout{No file \jobname.pre.}}
```

Result:

<details>

```
Rc files read:
  /home/user202729/.latexmkrc
  .latexmkrc
Latexmk: This is Latexmk, John Collins, 7 Jan. 2023. Version 4.79.
No existing .aux file, so I'll make a simple one, and require run of *latex.
Latexmk: applying rule 'lualatex'...
Rule 'lualatex':  Reasons for rerun
Category 'other':
  Rerun of 'lualatex' forced or previously required

------------
Run number 1 of rule 'lualatex'
------------
------------
Running 'lualatex  -recorder  "a.tex"'
------------
This is LuaHBTeX, Version 1.17.0 (TeX Live 2023/Arch Linux) 
 restricted system commands enabled.
(./a.tex
LaTeX2e <2022-11-01> patch level 1
 L3 programming layer <2023-02-22>
(/usr/share/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/usr/share/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texmf-dist/tex/latex/attachfile2/attachfile2.sty
(/usr/share/texmf-dist/tex/generic/iftex/iftex.sty)
(/usr/share/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texmf-dist/tex/latex/graphics/color.sty
(/usr/share/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/share/texmf-dist/tex/latex/graphics-def/luatex.def)
(/usr/share/texmf-dist/tex/latex/graphics/mathcolor.ltx))
(/usr/share/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
(/usr/share/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty)
(/usr/share/texmf-dist/tex/latex/kvoptions/kvoptions.sty
(/usr/share/texmf-dist/tex/latex/kvsetkeys/kvsetkeys.sty))
(/usr/share/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty)
(/usr/share/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/share/texmf-dist/tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/usr/share/texmf-dist/tex/latex/hycolor/hycolor.sty)
(/usr/share/texmf-dist/tex/latex/letltxmacro/letltxmacro.sty)
(/usr/share/texmf-dist/tex/latex/auxhook/auxhook.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texmf-dist/tex/latex/refcount/refcount.sty)
(/usr/share/texmf-dist/tex/generic/gettitlestring/gettitlestring.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/share/texmf-dist/tex/generic/intcalc/intcalc.sty)
(/usr/share/texmf-dist/tex/generic/etexcmds/etexcmds.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/puenc.def)
(/usr/share/texmf-dist/tex/latex/url/url.sty)
(/usr/share/texmf-dist/tex/generic/bitset/bitset.sty
(/usr/share/texmf-dist/tex/generic/bigintcalc/bigintcalc.sty))
(/usr/share/texmf-dist/tex/latex/base/atbegshi-ltx.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/hluatex.def
(/usr/share/texmf-dist/tex/generic/stringenc/stringenc.sty)
(/usr/share/texmf-dist/tex/latex/base/atveryend-ltx.sty)
(/usr/share/texmf-dist/tex/latex/rerunfilecheck/rerunfilecheck.sty
(/usr/share/texmf-dist/tex/generic/uniquecounter/uniquecounter.sty)))
(/usr/share/texmf-dist/tex/latex/attachfile2/atfi-luatex.def))
(/usr/share/texmf/tex/latex/asymptote/asymptote.sty
No file a.pre.
(/usr/share/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)))
(/usr/share/texmf-dist/tex/generic/iftex/ifpdf.sty)
(/usr/share/texmf-dist/tex/generic/iftex/ifxetex.sty)
(/usr/share/texmf-dist/tex/generic/catchfile/catchfile.sty))
(/usr/share/texmf-dist/tex/latex/l3backend/l3backend-luatex.def) (./a.aux)
(/usr/share/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/share/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty
(/usr/share/texmf-dist/tex/latex/grfext/grfext.sty)
(/usr/share/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))

Package asymptote Warning: file `a-1.pdf' not found on input line 9.

[1{/home/user202729/.texlive/texmf-var/fonts/map/pdftex/updmap/pdftex.map}]
(./a.aux)

Package rerunfilecheck Warning: File `a.out' has changed.
(rerunfilecheck)                Rerun to get outlines right
(rerunfilecheck)                or use package `bookmark'.

)
 411 words of node memory still in use:
   3 hlist, 1 vlist, 1 rule, 2 glue, 3 kern, 1 glyph, 4 attribute, 49 glue_spec
, 4 attribute_list, 1 write nodes
   avail lists: 2:49,3:9,4:12,5:39,6:3,7:67,9:54
</usr/share/texmf-dist/fonts/opentype/public/lm/lmroman10-regular.otf>
Output written on a.pdf (1 page, 2971 bytes).
Transcript written on a.log.
Latexmk: Getting log file 'a.log'
Latexmk: Examining 'a.fls'
Latexmk: Examining 'a.log'
Latexmk: Missing input file 'a.pre' (or dependence on it) from following:
  No file a.pre.
Latexmk: Missing input file 'a-1.pdf' (or dependence on it) from following:
  Package asymptote Warning: file `a-1.pdf' not found on input line 9.
Latexmk: References changed.
Latexmk: Log file says output to 'a.pdf'
Latexmk: applying rule 'cusdep asy pdf a-1'...
Rule 'cusdep asy pdf a-1':  Reasons for rerun
Changed files or newly in use/created:
  a-1.asy
Category 'no_dest':
  cusdep asy pdf a-1

------------
Run number 1 of rule 'cusdep asy pdf a-1'
------------
Latexmk: applying rule 'lualatex'...
Rule 'lualatex':  Reasons for rerun
Changed files or newly in use/created:
  a-1.pdf
  a.aux
  a.out
  a.pre

------------
Run number 2 of rule 'lualatex'
------------
------------
Running 'lualatex  -recorder  "a.tex"'
------------
This is LuaHBTeX, Version 1.17.0 (TeX Live 2023/Arch Linux) 
 restricted system commands enabled.
(./a.tex
LaTeX2e <2022-11-01> patch level 1
 L3 programming layer <2023-02-22>
(/usr/share/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/usr/share/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texmf-dist/tex/latex/attachfile2/attachfile2.sty
(/usr/share/texmf-dist/tex/generic/iftex/iftex.sty)
(/usr/share/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texmf-dist/tex/latex/graphics/color.sty
(/usr/share/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/share/texmf-dist/tex/latex/graphics-def/luatex.def)
(/usr/share/texmf-dist/tex/latex/graphics/mathcolor.ltx))
(/usr/share/texmf-dist/tex/generic/infwarerr/infwarerr.sty)
(/usr/share/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty)
(/usr/share/texmf-dist/tex/latex/kvoptions/kvoptions.sty
(/usr/share/texmf-dist/tex/latex/kvsetkeys/kvsetkeys.sty))
(/usr/share/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty)
(/usr/share/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/share/texmf-dist/tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/usr/share/texmf-dist/tex/latex/hycolor/hycolor.sty)
(/usr/share/texmf-dist/tex/latex/letltxmacro/letltxmacro.sty)
(/usr/share/texmf-dist/tex/latex/auxhook/auxhook.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texmf-dist/tex/latex/refcount/refcount.sty)
(/usr/share/texmf-dist/tex/generic/gettitlestring/gettitlestring.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/share/texmf-dist/tex/generic/intcalc/intcalc.sty)
(/usr/share/texmf-dist/tex/generic/etexcmds/etexcmds.sty)
(/usr/share/texmf-dist/tex/latex/hyperref/puenc.def)
(/usr/share/texmf-dist/tex/latex/url/url.sty)
(/usr/share/texmf-dist/tex/generic/bitset/bitset.sty
(/usr/share/texmf-dist/tex/generic/bigintcalc/bigintcalc.sty))
(/usr/share/texmf-dist/tex/latex/base/atbegshi-ltx.sty))
(/usr/share/texmf-dist/tex/latex/hyperref/hluatex.def
(/usr/share/texmf-dist/tex/generic/stringenc/stringenc.sty)
(/usr/share/texmf-dist/tex/latex/base/atveryend-ltx.sty)
(/usr/share/texmf-dist/tex/latex/rerunfilecheck/rerunfilecheck.sty
(/usr/share/texmf-dist/tex/generic/uniquecounter/uniquecounter.sty)))
(/usr/share/texmf-dist/tex/latex/attachfile2/atfi-luatex.def))
(/usr/share/texmf/tex/latex/asymptote/asymptote.sty (./a.pre)
(/usr/share/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)))
(/usr/share/texmf-dist/tex/generic/iftex/ifpdf.sty)
(/usr/share/texmf-dist/tex/generic/iftex/ifxetex.sty)
(/usr/share/texmf-dist/tex/generic/catchfile/catchfile.sty))
(/usr/share/texmf-dist/tex/latex/l3backend/l3backend-luatex.def) (./a.aux)
(/usr/share/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/share/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty
(/usr/share/texmf-dist/tex/latex/grfext/grfext.sty)
(/usr/share/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg)) (./a.out)
(./a.out)<<a-1.pdf>> [1{/home/user202729/.texlive/texmf-var/fonts/map/pdftex/up
dmap/pdftex.map}<./a-1.pdf>] (./a.aux))
 411 words of node memory still in use:
   3 hlist, 1 vlist, 1 rule, 2 glue, 3 kern, 1 glyph, 4 attribute, 49 glue_spec
, 4 attribute_list, 1 write nodes
   avail lists: 2:71,3:12,4:12,5:42,6:6,7:91,9:111
</usr/share/texmf-dist/fonts/opentype/public/lm/lmroman10-regular.otf>
Output written on a.pdf (1 page, 21300 bytes).
Transcript written on a.log.
Latexmk: Getting log file 'a.log'
Latexmk: Examining 'a.fls'
Latexmk: Examining 'a.log'
Latexmk: Log file says output to 'a.pdf'
Latexmk: All targets () are up-to-date
```

</details>
